retrofit/src/main/java/retrofit2/Retrofit.java

```diff
@@ -158,12 +159,14 @@ private void eagerlyValidateMethods(Class<?> service) {
     }
   }
 
-  ServiceMethod loadServiceMethod(Method method) {
-    ServiceMethod result;
+  ServiceMethod<?, ?> loadServiceMethod(Method method) {
+    ServiceMethod<?, ?> result = serviceMethodCache.get(method);
+    if (result != null) return result;
+
     synchronized (serviceMethodCache) {
```

Comment:
nit: weird to mix different sync mechanisms here. Not sure how to structure it better